pkg/controller/jobs/rayjob/rayjob_controller.go

```diff
@@ -103,16 +105,15 @@ func (j *RayJob) PodLabelSelector() string {
 }
 
 func (j *RayJob) PodSets() []kueue.PodSet {
-	// len = workerGroups + head
-	podSets := make([]kueue.PodSet, len(j.Spec.RayClusterSpec.WorkerGroupSpecs)+1)
+	podSets := make([]kueue.PodSet, 0)
```

Comment:
We can no longer used a fixed sized slice since the submitter Job PodSet is only added conditionally (if rayjob.Spec.SubmissionMode = K8sJobMode)

Reply:
rest of the code does not account for fixed size, so that's fine